master/usage/ipv6.md

```diff
@@ -1,31 +1,158 @@
+  hosts.
+  > **Note:** Each host running Calico currently must also have an IPv4
+  > address due to [bug 1204](https://github.com/projectcalico/calico/issues/1204).
+  {: .alert .alert-info}
+- Each host must have the sysctl setting `net.ipv6.conf.all.forwarding`
+  setting it to `1`.  This ensures both Kubernetes service traffic
+  and Calico traffic is forwarded appropriately.
+- Each host must have a default IPv6 route.
+
+#### Kubernetes Components Prerequisites
+
+Kubernetes components must be configured to operate with IPv6.
+Below are the flags for each Kubernetes component that will help ensure
+IPv6 operation.
+> **Note:** These flags are not the only flags that will need to be set on the
+> kubernetes components but the ones that were found necessary to ensure IPv6
+> operation.
+{: .alert .alert-info}
+- For kube-apiserver:
+  - The `--bind-address` or `--insecure-bind-address` flags should be set
+    to the appropriate IPv6 address or `::` for all IPv6 addresses on
+	the host.
+  - The `--advertise-address` flag should be set to the IPv6 address that
+    nodes should use to access the kube-apiserver.
+  - The `--service-cluster-ip-range` flag should be set to an IPv6 CIDR
+    that will be used for the Service IPs, the DNS service address must
+	be in this range.
+- For kube-controller-manager:
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - If the `--allocate-node-cidrs` flag is set then it is necessary to set
+    `--node-cidr-mask-size`. These are necessary when using host-local IPAM,
+    but if using calico-ipam is is easier to remove both flags as they are
+    not needed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+- For kube-scheduler:
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+- For kubelet:
+  - The `--address` flag should be set to the appropriate IPv6 address
+    or `::` for all IPv6 addresses.
+  - The `--cluster-dns` flag should be set to the IPv6 address that will
+    be used for the service DNS, this must be in the range used for
+	`--service-cluster-ip-range`.
+  - The `--node-ip` flag should be set to the IPv6 address of the node.
+- For kube-proxy:
+  - The `--bind-address` flag should be set to the appropriate IPv6
+    address or `::` for all IPv6 addresses on the host.
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+
+### Limitations
+
+- Currently Kubernetes supports only one IP stack version at a time. This
+  means that if you configure Kubernetes for IPv6 then {{site.prodname}}
+  should be configured to only assign IPv6 addresses.
+- You should not switch a deployment that is using IPv4 to IPv6, or
+  vise-versa but to instead setup a new cluster.
```

Comment:
"You should not switch a deployment that is using IPv4 to IPv6, or vise-versa but to instead setup a new cluster." --> "Do not switch a deployment using IPv4 to IPv6. Only enable IPv6 on a new cluster."

(btw, this should be almost the first sentence in the section)

Reply:
Did you see Casey's comment about changing this sentence? https://github.com/projectcalico/calico/pull/1736#discussion_r173524437
I like it since it is more positive about what this page is conveying.  Also more accurate about what we have tested ourselves since I did not actually try to take a deployment and switch it from IPv4 to IPv6.